aws/rust-runtime/aws-inlineable/src/rds_auth_token.rs

```diff
@@ -0,0 +1,284 @@
+            self.config.port(),
+            ACTION,
+            self.config.username()
+        );
+        let signable_request =
+            SignableRequest::new("GET", &url, std::iter::empty(), SignableBody::empty())
+                .expect("signable request");
+
+        let (signing_instructions, _signature) =
+            http_request::sign(signable_request, &signing_params.into())?.into_parts();
+
+        let mut url = url::Url::parse(&url).unwrap();
+        for (name, value) in signing_instructions.params() {
+            url.query_pairs_mut().append_pair(name, value);
+        }
+        let inner = url.to_string().split_off("https://".len());
+
+        Ok(AuthToken { inner })
+    }
+}
+
+/// Configuration for an RDS auth URL signer.
+#[derive(Debug, Clone)]
+pub struct Config {
+    /// The AWS credentials to sign requests with. Uses the default credential provider chain if not specified.
+    credentials: Option<SharedCredentialsProvider>,
+
+    /// The hostname of the database to connect to.
+    hostname: String,
+
+    /// The port number the database is listening on.
+    port: u32,
+
+    /// The region the database is located in. Uses the region inferred from the runtime if omitted.
+    region: Option<Region>,
+
+    /// The username to login as.
+    username: String,
+}
+
+impl Config {
+    /// Create a new `SignerConfigBuilder`.
+    pub fn builder() -> ConfigBuilder {
+        ConfigBuilder::default()
+    }
+
+    /// The AWS credentials to sign requests with.
+    pub fn credentials(&self) -> Option<&SharedCredentialsProvider> {
+        self.credentials.as_ref()
+    }
+
+    /// The hostname of the database to connect to.
+    pub fn hostname(&self) -> &str {
+        &self.hostname
+    }
+
+    /// The port number the database is listening on.
+    pub fn port(&self) -> u32 {
+        self.port
+    }
+
+    /// The region to sign requests with.
+    pub fn region(&self) -> Option<&Region> {
+        self.region.as_ref()
+    }
+
+    /// The DB username to login as.
+    pub fn username(&self) -> &str {
+        &self.username
+    }
+}
+
+/// A builder for [`Config`]s.
+#[derive(Debug, Default)]
+pub struct ConfigBuilder {
+    /// The AWS credentials to sign requests with. Uses the default credential provider chain if not specified.
+    credentials: Option<SharedCredentialsProvider>,
+
+    /// The hostname of the database to connect to.
+    hostname: Option<String>,
+
+    /// The port number the database is listening on.
+    port: Option<u32>,
+
+    /// The region the database is located in. Uses the region inferred from the runtime if omitted.
+    region: Option<Region>,
+
+    /// The database username to login as.
+    username: Option<String>,
+}
+
+impl ConfigBuilder {
+    /// Set the AWS credentials to sign requests with. Uses the default credential provider chain if not specified.
+    pub fn credentials(mut self, credentials: SharedCredentialsProvider) -> Self {
```

Comment:
Should the type of `credentials` be `impl ProvideCredentials` ([like this](https://github.com/smithy-lang/smithy-rs/blob/e6b154b03e9ab1d5b8ee723d6629724fb8f6f225/aws/rust-runtime/aws-config/src/lib.rs#L475-L477)) to not require callers to have shared ownership of it?